debezium-core/src/main/java/io/debezium/transforms/reselect/ReselectColumns.java

```diff
@@ -0,0 +1,300 @@
+ * value, this represents a value that wasn't changed nor was available in the change event. This will trigger
+ * this transformation to attempt to re-select the value from the target table and replace the unavailable
+ * value with the value currently in the database.
+ *
+ * Additionally, the user can specify a comma-separated list of columns that when the value of the column is
+ * {@code null}, the transformation will also re-query the target table and populate the field with the
+ * value currently in the database. This is useful for some connectors that may treat specific table columns
+ * like an unavailable value placeholder scenario but there is insufficient metadata at the JDBC level
+ * for Debezium to make sure determinations, such as Oracle Exadata extended strings.
+ *
+ * A specific connector should extend this class with its own implementation details
+ *
+ * @author Chris Cranford
+ */
+@Incubating
+public class ReselectColumns<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReselectColumns.class);
+
+    private static final Field COLUMN_LIST = Field.create("column.list")
+            .withDisplayName("List of columns to reselect")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.LONG)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .required()
+            .withDescription("Comma-separated list of columns to be reselected from source database");
+
+    private static final Field CONNECTION_URL = Field.create("connection.url")
+            .withDisplayName("JDBC connection url")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.LOW)
+            .required()
+            .withDescription("JDBC connection URL");
+
+    private static final Field CONNECTION_USER = Field.create("connection.user")
+            .withDisplayName("JDBC connection username")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.LOW)
+            .required()
+            .withDescription("JDBC connection username");
+
+    private static final Field CONNECTION_PASSWORD = Field.create("connection.password")
+            .withDisplayName("JDBC connection password")
+            .withType(ConfigDef.Type.PASSWORD)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.LOW)
+            .required()
+            .withDescription("JDBC connection password");
+
+    private static final Field UNAVAILABLE_VALUE_PLACEHOLDER = RelationalDatabaseConnectorConfig.UNAVAILABLE_VALUE_PLACEHOLDER;
+
+    protected SmtManager<R> smtManager;
+    protected List<Pattern> columnsList;
+    protected String placeholder;
+    protected ByteBuffer placeholderBytes;
+    protected Connection connection;
```

Comment:
I thought about this initially and opted not to go this route for a few reasons.

With raw JDBC, we can simply take the connection URL and credentials and open a connection during configure, which allows an immediate response of whether there is some configuration problem with the database connection.  

Using Debezium's JdbcConnection presents a few problems.  In Kafka Connect, there would only ever be one available, so we can simply iterate the collection, getting the only one returned and constructing the object.  If we don't find one because the connector doesn't implement it, we can log a failure so the user removes the SMT from the configuration.  

However, this gets trickier in Debezium Server because isn't there a single class loader that will have all the Debezium source connectors on the classpath?  This would result in multiple providers being available and then how would we know which one to use to establish a connection?

If you want, we can sync-up and discuss this in more detail after the team meeting on Tuesday.